### PR TITLE
Resurrect machine without allocation that are phoning home and alive

### PR DESCRIPTION
**Pull request by muhittink**

### CODE REVIEW

**Comment by Gerrit91** [comment]
Can you please give this PR a reasonable name that can then go into the release notes? Already forgot what this PR does exactly.

File: cmd/metal-api/internal/service/machine-service.go [hunk 1]
```diff
@@ -1806,7 +1806,10 @@ func ResurrectMachines(ds *datastore.RethinkStore, publisher bus.Publisher, ep *
 			continue
 		}
 
-		if provisioningEvents.Liveliness != metal.MachineLivelinessDead {
+		// sometimes a machine does not boot properly from the network and boots into the old operating system
+		// and then it emits "phoned home" events but has no allocation anymore
+		isPhoningHome := len(provisioningEvents.Events) > 0 && provisioningEvents.Liveliness == metal.MachineLivelinessAlive && provisioningEvents.Events[0].Event == metal.ProvisioningEventPhonedHome
```
Comment (majst01) [inline cmd/metal-api/internal/service/machine-service.go:1811]: This is clumsy

**Comment by majst01** [comment]
Current implementation is wrong, needs more thoughts how the final condition must look like

**Comment by Gerrit91** [comment]
With the FSM in place this will become easy to implement. Just resurrect if "FailedMachineReclaim" is true.

**Comment by Gerrit91** [comment]
Closed through https://github.com/metal-stack/metal-api/pull/146.

**Pull request closed without merging**